File: main/src/model/sleepGoal.java
package model;

public class sleepGoal extends Goal{

    private int currentAmount;
    private final int goalAmount;
    private final String goalType = "Hours";

    public sleepGoal(int goalAmount) {
        super();
        currentAmount = 0;
        this.goalAmount = goalAmount;
    }

    public void setCurrentAmount(int currentAmount) {
        this.currentAmount = currentAmount;
    }

    public int getCurrentAmount() {
        return currentAmount;
    }

    public int getGoalAmount() {
        return goalAmount;
    }

    public String getGoalType() {
        return goalType;
    }
}

File: main/src_js/sleepGoal.js
import Goal from "./Goal";
export default class sleepGoal extends Goal {
    constructor(goalAmount) {
        super();
        // private  currentAmount: number = 0;
        this.goalAmount = 0;
        this.goalType = "Hours";
        this.goalAmount = goalAmount;
    }
    // setCurrentAmount(currentAmount: number) {
    //     this.currentAmount = currentAmount;
    // }
    //
    // getCurrentAmount(): number {
    //     return this.currentAmount;
    // }
    getGoalAmount() {
        return this.goalAmount;
    }
    getGoalType() {
        return this.goalType;
    }
}